specification/sections/02-providers.md

```diff
@@ -203,12 +203,15 @@ title: Provider State
 stateDiagram-v2
     direction LR
     [*] --> NOT_READY
-    NOT_READY --> READY
+    NOT_READY --> READY:initialize
     READY --> ERROR
     ERROR --> READY
     READY --> STALE
     STALE --> READY
     STALE --> ERROR
+    READY --> NOT_READY:shutdown
+    STALE --> NOT_READY:shutdown
+    ERROR --> NOT_READY:shutdown
```

Comment:
I added names for a few of the transitions because it made the diagram less confusing. Adding them everywhere seemed redundant though.

Reply:
The preview looks good. IMO good improvement to explain _what_ cause the state change.

```mermaid
---
title: Provider State
---
stateDiagram-v2
    direction LR
    [*] --> NOT_READY
    NOT_READY --> READY:initialize
    READY --> ERROR
    ERROR --> READY
    READY --> STALE
    STALE --> READY
    STALE --> ERROR
    READY --> NOT_READY:shutdown
    STALE --> NOT_READY:shutdown
    ERROR --> NOT_READY:shutdown
```